### ISSUE TITLE
MutableNetwork.hasEdgeConnecting throw IllegalArgumentException when node are missing

### ISSUE THREAD

**Issue by fathzer** [google/guava#3721]
According to the javadoc, the hasEdgeConnecting method is equivalent to nodes().contains(nodeU) && successors(nodeU).contains(nodeV).
But, unlike this expression, it throws an exception if the nodes are unknown.
Here is a simple demonstration
```java
import com.google.common.graph.MutableNetwork;
import com.google.common.graph.NetworkBuilder;

public class Bug {
	public static void main(String[] args) {
		final MutableNetwork<String, String> graph = NetworkBuilder.undirected().allowsParallelEdges(false).build();
		// Foolowing line prints false
		System.out.println(graph.nodes().contains("1") && graph.successors("1").contains("2"));
		// Next should print false but throws an IllegalArgumentException
		System.out.println(graph.hasEdgeConnecting("1", "2"));
	}
}
```
version guava-28.1-jre

**Comment by nglorioso** [comment]
It looks like a bit subtle, but `successors(nodeU).contains(nodeV)` throws an `IllegalArgumentException` when `nodeU` is not in the network (which is the actual behavior you're seeing here).

However, you're right that the short-circuiting behavior of `&&` isn't replicating here. That seems like an important part of the documented behavior :). FWIW, it appears to also throw an IllegalArgumentException even if `"1"` is in the network but `"2"` isn't.

**Comment by jrtom** [comment]
This is just a straight-up bug on our parts; I'm coding up a fix now.

The problem originates in the fact that `hasEdgeConnecting()` is actually (in effect) calling `edgesConnecting().isEmpty()`.  This seems reasonable on the face of it, but `edgesConnecting(u, v)` has the contract that it throws if either `u` or `v` are not in the graph; this is consistent with the contracts for the other `Set`-returning accessors, but it's not what we want for boolean-returning queries like `hasEdgeConnecting()`.

Sorry about that; thanks for the catch.  :)

**Comment by jrtom** [comment]
I've submitted an internal fix; this should be propagating externally soon.

**Comment by jrtom** [comment]
This is fixed in v28.2: https://github.com/google/guava/releases/tag/v28.2
